### PR TITLE
Flip Y-Axis of X-Rays to get a right handed coordinate system.

### PR DESCRIPTION
**Pull request by SirVer**
This means that the images origin is now at the bottom left of the images, i.e. the images also use a right handed coordinate system.

Also a grab bag of unrelated other small refactorings:
- Unifying the way the two javascript directories are build.
- Add a few missing functions that are in the octree, but not the quadtree crate.
- Make the lowest level of the X-Rays use nearest interpolation, so that the actual resolution of the X-Rays is visible.
- Bump the X-Ray quadtree version, since the tiles are flipped.

### CODE REVIEW

File: octree_web_viewer/client/octree_viewer.ts [hunk 1]
```diff
@@ -247,6 +247,10 @@ class NodeData {
   }
 }
 
+function now(): number {
+  return +new Date();
```
Comment (pdubroy) [inline octree_web_viewer/client/octree_viewer.ts]: Same as above, you can use `performance.now()`.

File: octree_web_viewer/client/package.json [hunk 2]
```diff
@@ -4,18 +4,23 @@
...
     "prettier": "^1.10.2",
+    "rollup": "^0.55.3",
+    "rollup-plugin-commonjs": "^8.3.0",
+    "rollup-plugin-node-resolve": "^3.0.2",
+    "typescript": "=2.7.2"
+  },
+  "dependencies": {
     "@types/dat-gui": "0.6.1",
-    "@types/three": "0.0.27",
+    "@types/three": "0.92.17",
```
Comment (pdubroy) [inline octree_web_viewer/client/package.json]: Shouldn't the minor version match the version of three.js you're using?

File: xray/src/generation.rs [hunk 3]
```diff
@@ -272,6 +272,50 @@ impl HeightStddevColoringStrategy {
     }
 }
 
+/// Build a parent image created of the 4 children tiles. All tiles are optionally, in which case
```
Comment (pdubroy) [inline xray/src/generation.rs]: Nit: why three slashes? Is it a Rust thing? I've never seen that before.

File: octree_web_viewer/client/main.ts [hunk 4]
```diff
@@ -14,10 +14,11 @@
 
 'use strict';
 
+import * as THREE from 'three';
 import { FirstPersonController } from './control';
 import { OctreeViewer } from './octree_viewer';
 
-export function now(): number {
+function now(): number {
   return +new Date();
```
Comment (pdubroy) [inline octree_web_viewer/client/main.ts]: `new Date().getTime()` would be clearer here. But you can also just use `performance.now()`.

File: octree_web_viewer/client/package.json (hunk 2 above)
Comment (SirVer) [reply octree_web_viewer/client/package.json]: Dunno. I copied that from our internal repositories. 0.95 does not exist for the types, so it seems I need to do this somehow.

File: xray/src/generation.rs (hunk 3 above)
Comment (SirVer) [reply xray/src/generation.rs]: https://doc.rust-lang.org/beta/rustdoc/what-is-rustdoc.html

Rust has something like doxygen build in.

**Comment by SirVer** [comment]
> lgtm with some suggestions. Also, shouldn't the PR title say "y-axis" and not "x-axis"?

No, because we align X with X. To get a right handed coordinate system this means we align Y with minus Y.

**Comment by SirVer** [comment]
@wally-the-cartographer merge

**Pull request merged**